client/pages/chat-page.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+export default class ChatMain extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      chatMsg: null
+    };
+  }
+
+  componentDidMount() {
+    // const socket = io();
+    // socket.on('connection');
+  }
+
```

Comment:
Be sure to store the socket so that you can disconnect when the user leaves the room

```js
componentDidMount() {
  this.socket = io({ query: `roomName=${this.props.roomName}` });
}

componentWillUnmount() {
  this.socket.disconnect();
}
```